README.md

```diff
@@ -0,0 +1,76 @@
+# Nexmo JWT Generator for Ruby
+
+[![Gem Version](https://badge.fury.io/rb/nexmo-jwt.svg)](https://badge.fury.io/rb/nexmo-jwt) [![Build Status](https://api.travis-ci.org/Nexmo/nexmo-jwt-ruby.svg?branch=master)](https://travis-ci.org/Nexmo/nexmo-jwt-ruby) [![Coverage Status](https://coveralls.io/repos/github/Nexmo/nexmo-jwt-ruby/badge.svg?branch=coveralls)](https://coveralls.io/github/Nexmo/nexmo-jwt-ruby?branch=master)[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-v2.0%20adopted-ff69b4.svg)](CODE_OF_CONDUCT.md)
+
+<img src="https://developer.nexmo.com/assets/images/Vonage_Nexmo.svg" height="48px" alt="Nexmo is now known as Vonage" />
+
+This is the Ruby library to generate Nexmo JSON Web Tokens (JWTs). To use it you'll
+need a Nexmo account. Sign up [for free at nexmo.com][signup].
+
+* [Requirements](#requirements)
+* [Installation](#installation)
+* [Usage](#usage)
+* [Documentation](#documentation)
+* [License](#license)
+
+## Requirements
+
+The JWT generator supports Ruby version 2.5 or newer.
+
+## Installation
+
+To install the Ruby client library using Rubygems:
+
+    gem install nexmo-jwt
+
+Alternatively you can clone the repository:
+
+    git clone git@github.com:Nexmo/nexmo-jwt-ruby.git
+
+## Usage
+
+By default the Nexmo JWT generator creates a short lived JWT (15 minutes) per request.
+To generate a long lived JWT for multiple requests, specify a longer value in the `exp`
+parameter in the `builder` block. 
+
+Example with no custom configuration:
+
+```ruby
+token = Nexmo::JWTBuilder.build(application_id = YOUR_APPLICATION_ID, private_key = YOUR_PRIVATE_KEY)
+```
+
+Example using the `builder` block to provide custom configuration options:
+
+```ruby
+token = Nexmo::JWTBuilder.build(application_id = YOUR_APPLICATION_ID, private_key = YOUR_PRIVATE_KEY) do |builder|
+  builder.set_exp 500
```

Comment:
This doesn't feel Ruby-like to me. Should it be done with accessors?

```
builder.exp = 500
```

Using methods to set values only really applies in languages where you can't have methods with the same name as an instance variable

Reply:
I specifically followed the specification for the new libraries which recommends this design pattern. You are right, it is not particularly common, but it is also not entirely uncommon either.

Reply:
My gut feel is that this should be an exception to the spec based on langugage.

See https://github.com/rubocop-hq/ruby-style-guide#accessormutator-method-names from Rubocop

Reply:
I hear you. I can go back to the drawing board and rewrite this again. For this 2nd iteration, I tried to follow the agreed-upon specification, the first time was closely following the already implemented JWT utility in the SDK. If I am hearing you correctly, for the third time, we are to build with the spec but not follow it exactly?